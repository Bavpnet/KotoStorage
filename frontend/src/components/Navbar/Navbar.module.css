@value mobile-max-width from './../../globals.css';
@value primary from './../../globals.css';

nav {
  font-family: 'Gudea', sans-serif;
  background-color: transparent;
  display: flex;
  justify-content: center;
  align-items: center;
  gap: min(300px, 10%);
  overflow: hidden;
  padding-top: 32px;
}

aside.active {
  left: 0;
}

.logo {
  font-size: 1rem;
  font-weight: 700;
}

.gh_btn {
  cursor: pointer;
}

@media only screen and (max-width: mobile-max-width) {
  nav {
    justify-content: space-between;
    padding: 18px 20px;
  }

  nav > *:not(aside) {
    z-index: 10;
  }

  aside {
    position: fixed;
    top: 0;
    left: 100%;
    width: 100%;
    height: 100vh;
    display: flex;
    flex-direction: column;
    background: primary;
    padding-top: 64px;
    gap: 52px;
    z-index: 5;
    transition: 0.4s;
  }

  .gh_btn {
    display: none;
  }
}
