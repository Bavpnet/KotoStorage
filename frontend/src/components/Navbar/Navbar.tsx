import { useToggle } from '../../hooks/useToggle'
import { LinkButtonGroup } from '../LinkButtonGroup'
import { SocialInfo } from '../SocialInfo'
import { Hamburger } from '../ui/Hamburger'
import { UIAnchor } from '../ui/UIAnchor'
import styles from './Navbar.module.css'

export const Navbar = () => {
  const [isMenuOpen, toggleMenu] = useToggle(false)

  return (
    <nav>
      <h1 className={styles.logo}>KotoStorage</h1>

      <aside className={`${isMenuOpen ? styles.active : ''}`}>
        <LinkButtonGroup isMenuOpen={isMenuOpen} toggleMenu={toggleMenu} />
        <SocialInfo isFooterInfo={false} />
      </aside>

      <div className={styles.ghBtn}>
        <UIAnchor
          text={'GitHub'}
          href="https://github.com/Bavpnet/KotoStorage"
          target="_blank"
        />
      </div>

      <Hamburger onClick={toggleMenu} isMenuOpen={isMenuOpen} />
    </nav>
  )
}
