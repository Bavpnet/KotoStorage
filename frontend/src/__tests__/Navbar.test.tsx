import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { Navbar } from '../components/Navbar'

describe('Navbar', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Navbar />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
