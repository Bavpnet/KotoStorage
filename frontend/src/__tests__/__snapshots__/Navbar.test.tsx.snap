// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav>
  <h1
    className="logo"
  >
    KotoStorage
  </h1>
  <aside
    className=""
  >
    <ul
      className="links"
    >
      <li
        className="line"
        role="presentation"
      />
      <li>
        <button
          className="linksBtn "
          onClick={[Function]}
        >
          <span>
            About
          </span>
        </button>
      </li>
      <li>
        <button
          className="linksBtn "
          onClick={[Function]}
        >
          <span>
            Upload
          </span>
        </button>
      </li>
      <li>
        <button
          className="linksBtn "
          onClick={[Function]}
        >
          <span>
            Gallery
          </span>
        </button>
      </li>
      <li
        className="line"
        role="presentation"
      />
    </ul>
    <section
      className="inSidebar"
    >
      <h2
        className="infoTitle"
      >
        Follow us in the media
      </h2>
      <p
        className="infoDesc"
      >
        Hear about exclusive news and changes in the cats world
      </p>
      <ul
        className="social"
      >
        <li>
          Instagram
        </li>
        <li>
          Telegram
        </li>
        <li>
          Github
        </li>
      </ul>
    </section>
  </aside>
  <div
    className="ghBtn"
  >
    <a
      className="anc"
      href="https://github.com/Bavpnet/KotoStorage"
      target="_blank"
    >
      GitHub
    </a>
  </div>
  <button
    className="hamburger "
    onClick={[Function]}
  >
    <span
      className="bar"
    />
    <span
      className="bar"
    />
    <span
      className="bar"
    />
  </button>
</nav>
`;
